### PR TITLE
Add cut Doublebarrel shotgun by using model from Titanfall Online

### PR DESCRIPTION
**Pull request by EM4Volts**
Video ingame:
https://cdn.discordapp.com/attachments/1121520467791056997/1124063836702523422/yeet.mp4

all stats have been brought over from TFO aswell.
should make people more happy when playing neinguars gungame as it features said weapon but used the weapons placeholder model (a mastiff).

Replace this line with a description of your change (and screenshots/screenrecordings if applicable).

reload speed in video is slower in vid then ingame as video was made after the reload was made to fit the tfo reload

### CODE REVIEW

**Comment by EM4Volts** [comment]
all materials have now been packed into a vpk,
as for renaming the rpak i dont know if that is going to be a good idea
because recently someone tried extracting one of my rpaks and the images inside were a bit broken/shifted afterwards so i think seperate might be better rn.

**Comment by ASpoonPlaysGames** [comment]
> Going by current structure `.vtf` and `.vmt` fies should be packed in the vpk

No? Those have been able to be loaded from disk for a while (and mod settings doesnt pack its vmt and vtf into vpk)

**Comment by F1F7Y** [comment]
> No? Those have been able to be loaded from disk for a while (and mod settings doesnt pack its vmt and vtf into vpk)

I know we can ship these without packing them into a vpk, but so far all materials have been packed into `client_mp_northstar_custom.vpk`.

We either ship everything in the vpk or we ship everything outside as separate files.

**Comment by ASpoonPlaysGames** [comment]
> I know we can ship these without packing them into a vpk, but so far all materials have been packed into `client_mp_northstar_custom.vpk`.

https://github.com/R2Northstar/NorthstarMods/tree/main/Northstar.Client/mod/materials/vgui would like a word

**Comment by ASpoonPlaysGames** [comment]
But also, those materials were added to the vpk because at the time we couldnt load vmts from disk.

**Comment by EM4Volts** [comment]
should i leave the vmt/vft inside of the vpk now or change it to be outside again?

**Comment by ASpoonPlaysGames** [comment]
> should i leave the vmt/vft inside of the vpk now or change it to be outside again?

Just leave it for now. I might make a separate PR to move everything outside of VPKs (in repo and then put them in VPKs in CI) but that's beyond the scope of this PR.

**Comment by EM4Volts** [comment]
> > should i leave the vmt/vft inside of the vpk now or change it to be outside again?
> 
> Just leave it for now. I might make a separate PR to move everything outside of VPKs (in repo and then put them in VPKs in CI) but that's beyond the scope of this PR.

ok, as said, didnt quite know wether im now supposed to put them in a vpk or not lol

**Review by F1F7Y — CHANGES_REQUESTED** [review]
I'm fine with things as they are rn, but would prefer if the rpak was named after the gun script ( eg `mp_weapon_shotgun_doublebarrel` )

**Comment by ASpoonPlaysGames** [comment]
> I'm fine with things as they are rn, but would prefer if the rpak was named after the gun script ( eg `mp_weapon_shotgun_doublebarrel` )

Fair warning, if you do this rebuild the rpak with the new name. Just renaming the file won't work (it'll try to load the starpak with the original name)

**Review by F1F7Y — CHANGES_REQUESTED** [review]
After going through this once again:
1. The path to the mdl should be `models/weapons/weapon_shotgun_doublebarrel/`
2. The name of the mdl should be `ptpov_shotgun_doublebarrel.mdl`
3. The name of the rpak should be `weapon_shotgun_doublebarrel.rpak` ( or something similar to the script name

~~This is also missing a call to `PrecacheWeapon` in `ns_northstar_custom_precache.gnut`~~
And current version as of this review is crashing due to no `rpak.json`

Most of these names i based off of other custom weapons.

**Comment by ASpoonPlaysGames** [comment]
Wait we don't have the third person model? :(

**Comment by ASpoonPlaysGames** [comment]
> 3\. The name of the rpak should be `weapon_shotgun_doublebarrel.rpak` ( or something similar to the script name

For the sake of readability, just keep it identical to script name (`mp_weapon_shotgun_doublebarrel.rpak`)

**Comment by EM4Volts** [comment]
rpaks should be named correctly in my doublebarrel branch now, also added a world model

**Comment by F1F7Y** [comment]
> rpaks should be named correctly in my doublebarrel branch now, also added a world model

could you also rename the folder and the mdls like i suggested in my last review

**Comment by F1F7Y** [comment]
> the foldername is the same as in tfo, thats why i kept it.

we're not Nexon ( thankfully )

**Review by F1F7Y — COMMENTED** [review]
Tested together with #600  on Launcher [#504](https://github.com/R2Northstar/NorthstarLauncher/pull/504)

Works

Will approve once the folders and mdls are renamed

**Comment by EM4Volts** [comment]
rename done

**Comment by p0358** [comment]
Does it belong to the main mod though? Sounds like an ideal use-case for the fancy mod auto-download system that they were working on innit

**Comment by F1F7Y** [comment]
> Does it belong to the main mod though? Sounds like an ideal use-case for the fancy mod auto-download system that they were working on innit

I think you confused your nationality, instead of "innit" you should have ended the sentence with "xd"

To answer your question you can make the same point for everything in Northstar.Custom. Including it in Northstar means more people will be able to use it without having to figure out how mods work. Second point would be that auto-download is stuck in PR hell ig

**Review by F1F7Y — APPROVED** [review]
Works

**Comment by F1F7Y** [comment]
@GeckoEidechse adds vanilla file `mp_weapon_shotgun_doublebarrel.txt` from mp_common

**Comment by p0358** [comment]
> I think you confused your nationality, instead of "innit" you should have ended the sentence with "xd"

shut up xd

**Comment by EM4Volts** [comment]
was tested with others on desynced servers (servers without the mod)

crashes users with mod if someone without it holds it in hands.

**Review by GeckoEidechse — APPROVED** [review]
Compatibility in matrix

|                           | old server  | server with double barrel |
| ------------------------- | ----------- | ------------------------- |
| old client                | works       | error mdl model           |
| client with double barrel | client crashes for some reason but only when someone has the gun equipped | works |

File: Northstar.Custom/mod/scripts/weapons/mp_weapon_shotgun_doublebarrel.txt
Comment (GeckoEidechse) [inline Northstar.Custom/mod/scripts/weapons/mp_weapon_shotgun_doublebarrel.txt:1]: Waiting on https://github.com/EM4Volts/NorthstarMods/pull/1

**Comment by GeckoEidechse** [comment]
So while looking through this I noticed that original version had segmented reload which honestly I think it makes sense for this gun. Now the issue is that I have no idea how to get that working. Tried to use some of the original values but without success :/

**Review by F1F7Y — APPROVED** [review]
Works like a gun

**Comment by EM4Volts** [comment]
> So while looking through this I noticed that original version had segmented reload which honestly I think it makes sense for this gun. Now the issue is that I have no idea how to get that working. Tried to use some of the original values but without success :/

i attempted todo so but i couldnt get it to work, tried replicating the way the mastiff does it but something is borked i guess.

**Review by H0L0theBard — APPROVED** [review]
LGTM (Looks gun to me) :)

**Comment by GeckoEidechse** [comment]
I'd say we do like a playtest with 2-3 people again tonight and then merge

**Review by GeckoEidechse — APPROVED** [review]
Re-tested in private match with @uniboi. Gun is a bit OP in damage but values are copied (and reduced) from TFO so I guess it's fine ¯\\\_(ツ)\_/¯

**Pull request merged**